bugsnag-android-core/src/main/java/com/bugsnag/android/Configuration.java

```diff
@@ -91,7 +91,33 @@ public Configuration(@NonNull String apiKey) {
             autoDetectNdkCrashes = false;
         }
 
-        loggingEnabled = !AppData.RELEASE_STAGE_PRODUCTION.equals(releaseStage);
+        boolean loggingEnabled = !AppData.RELEASE_STAGE_PRODUCTION.equals(releaseStage);
+
+        if (loggingEnabled) {
```

Comment:
I think doing this logic in the constructor may be error-prone… what happens if a `Configuration` object is instantiated, the the developer calls `config.setReleaseStage(str)`. Does this get re-evalutated?

Reply:
If the developer calls `config.setReleaseStage()` the `logger` implementation does not currently change.

I think it would be best to avoid these sorts of side-effects in setters, but this may be something we want to discuss more broadly if it's not already mentioned in the spec.

Reply:
Yeah I think doing it as a side effect in setters would be a complex and error-prone way to implement it.

I think it's important to fix this though – I'd consider it a bug, but at best it's difficult to explain: "in production we will switch off logging, except if you set releaseStage=production in a particular way"

I would use the following steps to achieve consistent logic, and I'd place in the client constructor:

```
if config.logger is set by the developer
  this.logger = config.logger
else
  if config.releaseStage is 'production'
    this.logger = NoopLogger
  else
    this.logger = DebugLogger

Reply:
I've updated this so that if the `releaseStage` has been set to 'production' after `Configuration` has been constructed, the `Client` will use a `DebugLogger` if `config.logger` hasn't been overridden by the user.